test/e2e/empty_dir_wrapper.go

```diff
@@ -0,0 +1,179 @@
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/util/intstr"
+
+	"strconv"
+
+	. "github.com/onsi/ginkgo"
+)
+
+// This test will create a pod with a secret volume and gitRepo volume
+// Thus requests a secret, a git server pod, and a git server service
+var _ = Describe("EmptyDir wrapper volumes", func() {
+	f := NewFramework("secrets")
+
+	It("should becomes running [Conformance]", func() {
+		name := "secret-test-" + string(util.NewUUID())
+		volumeName := "secret-volume"
+		volumeMountPath := "/etc/secret-volume"
+
+		secret := &api.Secret{
+			ObjectMeta: api.ObjectMeta{
+				Namespace: f.Namespace.Name,
+				Name:      name,
+			},
+			Data: map[string][]byte{
+				"data-1": []byte("value-1\n"),
+			},
+		}
+
+		var err error
+		if secret, err = f.Client.Secrets(f.Namespace.Name).Create(secret); err != nil {
+			Failf("unable to create test secret %s: %v", secret.Name, err)
+		}
+
+		gitServerPodName := "git-server-" + string(util.NewUUID())
+		containerPort := 8000
+
+		labels := map[string]string{"name": gitServerPodName}
+
+		gitServerPod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name:   gitServerPodName,
+				Labels: labels,
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:            "git-repo",
+						Image:           "resouer/fakegitserver:v1",
+						ImagePullPolicy: "IfNotPresent",
+						Ports: []api.ContainerPort{
+							{ContainerPort: containerPort},
+						},
+					},
+				},
+			},
+		}
+
+		if gitServerPod, err = f.Client.Pods(f.Namespace.Name).Create(gitServerPod); err != nil {
+			Failf("unable to create test git server pod %s: %v", gitServerPod.Name, err)
+		}
+
+		// Portal IP and port
+		httpPort := 2345
+
+		gitServerSvc := &api.Service{
+			ObjectMeta: api.ObjectMeta{
+				Name: "git-server-svc",
+			},
+			Spec: api.ServiceSpec{
+				Selector: labels,
+				Ports: []api.ServicePort{
+					{
+						Name:       "http-portal",
+						Port:       httpPort,
+						TargetPort: intstr.FromInt(containerPort),
+					},
+				},
+			},
+		}
+
+		if gitServerSvc, err = f.Client.Services(f.Namespace.Name).Create(gitServerSvc); err != nil {
+			Failf("unable to create test git server service %s: %v", gitServerSvc.Name, err)
+		}
+
+		gitVolumeName := "git-volume"
+		gitVolumeMountPath := "/etc/git-volume"
+		gitURL := "http://" + gitServerSvc.Spec.ClusterIP + ":" + strconv.Itoa(httpPort)
+		gitRepo := "test"
+
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name: "pod-secrets-" + string(util.NewUUID()),
+			},
+			Spec: api.PodSpec{
+				Volumes: []api.Volume{
+					{
+						Name: volumeName,
+						VolumeSource: api.VolumeSource{
+							Secret: &api.SecretVolumeSource{
+								SecretName: name,
+							},
+						},
+					},
+					{
+						Name: gitVolumeName,
+						VolumeSource: api.VolumeSource{
+							GitRepo: &api.GitRepoVolumeSource{
+								Repository: gitURL,
+								Directory:  gitRepo,
+							},
+						},
+					},
+				},
+				Containers: []api.Container{
```

Comment:
Yes, otherwise, kubelet will report "volume path not found", and this pod will never becomes running.